src/lib/components/Subscribe/index.js

```diff
@@ -0,0 +1,109 @@
+/**
+ * @fileoverview Element that renders newsletter subscription form.
+ */
+
+import {BaseElement} from "../BaseElement";
+import "./_styles.scss";
+
+/**
+ * Element that renders newsletter subscription form.
+ *
+ * @extends {BaseElement}
+ * @final
+ */
+class Subscribe extends BaseElement {
+  constructor() {
+    super();
+
+    this.form = this.querySelector("form");
+    this.subscribeError = this.querySelector("div#w-subscribe__error");
+    this.subscribeMessage = this.querySelector("p#w-subscribe__message");
+    this.checkboxes = ["WebDevNewsletter", "collects-pii-spii-checkbox"];
+    this.needsDoubleOptIn = [
+      "AT: Austria",
+      "DE: Germany",
+      "GR: Greece",
+      "LU: Luxembourg",
+      "NO: Norway",
+    ];
+    this.robotName = "is-it-just-me-or-was-this-form-filled-out-by-a-robot";
+    this.submissionUrl = this.form.action;
+    this.processing = false;
+    this.submitted = false;
+
+    this.form.addEventListener("submit", this.onSubmit.bind(this));
+  }
+
+  /**
+   * @param {FormData} form
+   * @return {FormData}
+   */
+  cleanForm(form) {
+    const doubleOptIn = this.needsDoubleOptIn.includes(form.get("Country"));
+    this.checkboxes.forEach((checkbox) =>
+      form.set(checkbox, doubleOptIn ? "Unconfirmed" : "True"),
+    );
+    form.delete(this.robotName);
+    return form;
+  }
+
+  postForm(body) {
+    return fetch(this.submissionUrl, {
+      method: "POST",
+      body,
+    }).then((r) => r.json());
+  }
+
+  onError(errors) {
+    this.subscribeError.innerHTML = "";
+    if (errors) {
+      Object.values(errors).forEach((e) => {
+        const pTag = document.createElement("p");
+        pTag.innerText = typeof e === "string" ? e : e.join(" ");
```

Comment:
This seems like an area where having slots and shadow DOM would be helpful because then we could have LitElement render this for us...

I'm anxious about creating DOM on the fly and inserting it into the page, especially if the text we're inserting is coming from an external service. Feels like an XSS vector. I don't know if it's overkill but one option would be to make `w-subscribe__error` into a little custom element that just renders whatever array of errors we give it. @samthor — any better suggestions?

Reply:
Also `.innerText` is harmful. Don't use it- use `.textContent` instead.

This isn't XSS since we're not setting `.innerHTML`.

I think I don't want to block Micheal from getting this PR in. I think we revisit the Shadow DOM approach soon and render in lit with its in-built array support.